python-sdk/dev/scripts/pre_commit_readme_sync.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python
+from __future__ import annotations
+
+import importlib
+from pathlib import Path
+from typing import Sequence
+
+SOURCES_ROOT = Path(__file__).parents[2]
+ASTRO_ROOT = SOURCES_ROOT / "src" / "astro"
+README_PATH = SOURCES_ROOT / ".." / "README.md"
+
+HEADING = "## Supported technologies"
+
+
+def get_constants_module():
+    """
+    Get constants module
+    """
+    spec = importlib.util.spec_from_file_location("constants", f"{ASTRO_ROOT}/constants.py")
+    mod = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(mod)
+    return mod
+
+
+# Translation dictionaries for table alignment
+left_rule = {"<": ":", "^": ":", ">": "-"}
+right_rule = {"<": "-", "^": ":", ">": ":"}
+
+
+def evalute_field(record, field_spec):
+    """
+    Evaluate a field of a record using the type of the field_spec as a guide.
+    """
+    if type(field_spec) is int:
+        return str(record[field_spec])
+    elif type(field_spec) is str:
+        return str(getattr(record, field_spec))
+    else:
+        return str(field_spec(record))
+
+
+def get_table(records, fields, headings, alignment=None):
```

Comment:
@kaxil Is there a better way to create a markdown table? 
I tried with one lib but the env we execute the hook doesn't have that module installed.